src/ComServerHelpers/WinRtServer.cs

```diff
@@ -0,0 +1,295 @@
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Marshalling;
+using System.Runtime.Versioning;
+using System.Threading.Tasks;
+using ComServerHelpers.Internal;
+using ComServerHelpers.Internal.Windows;
+using Windows.Win32.Foundation;
+using Windows.Win32.System.Com;
+using Windows.Win32.System.WinRT;
+using WinRT;
+using static Windows.Win32.PInvoke;
+//using unsafe DllActivationCallback = delegate* unmanaged[Stdcall]<Windows.Win32.System.WinRT.HSTRING, Windows.Win32.System.WinRT.IActivationFactory**, Windows.Win32.Foundation.HRESULT>;
+
+namespace ComServerHelpers;
+
+/// <summary>
+/// An Out of Process Windows Runtime Server.
+/// </summary>
+/// <remarks>
+/// <para>Allows for types to be created using WinRT activation instead of COM activation like <see cref="ComServer"/>.</para>
+/// <para>Typical usage is to call from an <see langword="await"/> <see langword="using"/> block, using <see cref="WaitForFirstObjectAsync"/> to not close until it is safe to do so.</para>
+/// <code language="cs">
+/// <![CDATA[
+/// using (WinRtServer server = new WinRtServer())
+/// {
+///     server.RegisterClass<RemoteThing>();
+///     server.Start();
+///     await server.WaitForFirstObjectAsync();
+/// }
+/// ]]>
+/// </code>
+/// </remarks>
+/// <see cref="IDisposable"/>
+/// <threadsafety static="true" instance="false"/>
+[SupportedOSPlatform("windows8.0")]
+[System.Diagnostics.CodeAnalysis.SuppressMessage("Naming", "CA1724", Justification = "No better idea")]
+public sealed class WinRtServer : IDisposable
+{
+    /// <summary>
+    /// Mapping of Activatable Class IDs to activation factories and their <see cref="ComWrappers"/> implementation.
+    /// </summary>
+    private readonly Dictionary<string, (BaseActivationFactory Factory, ComWrappers Wrapper)> factories = [];
+
+    private readonly unsafe DllGetActivationFactory activationFactoryCallbackWrapper;
+
+    private unsafe readonly delegate* unmanaged[Stdcall]<HSTRING, IActivationFactory**, HRESULT> activationFactoryCallbackPointer;
+    private readonly StrategyBasedComWrappers comWrappers = new();
+
+    /// <summary>
+    /// Tracks the creation of the first instance after server is started.
+    /// </summary>
+    private TaskCompletionSource<object>? firstInstanceCreated;
+
+    private RO_REGISTRATION_COOKIE registrationCookie = (RO_REGISTRATION_COOKIE)0;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="WinRtServer"/> class.
+    /// </summary>
+    public unsafe WinRtServer()
+    {
+        activationFactoryCallbackWrapper = ActivationFactoryCallback;
+        activationFactoryCallbackPointer = (delegate* unmanaged[Stdcall]<HSTRING, IActivationFactory**, HRESULT>)Marshal.GetFunctionPointerForDelegate(activationFactoryCallbackWrapper);
+
+        HRESULT result = RoInitialize(RO_INIT_TYPE.RO_INIT_MULTITHREADED);
```

Comment:
`RoInitialize` is called by the .NET runtime at thread startup. Why do you need to call it here?

Also, this will fail if someone puts `[STAThread]` on the Main method and calls this from the main thread or starts a thread as an STA thread. Is that intentional? If so, I think this should specifically check the apartment and throw a better exception or have at least a comment about it.

Reply:
I don't think we need this at all.